relay-metrics/src/bucket.rs

```diff
@@ -556,10 +565,105 @@ impl From<MetricValue> for BucketValue {
     }
 }
 
-/// Error returned when parsing or serializing a [`Bucket`].
-#[derive(Debug, Error)]
-#[error("failed to parse metric bucket")]
-pub struct ParseBucketError(#[source] serde_json::Error);
+/// Parses a list of counter values separated by colons and sums them up.
+fn parse_counter(string: &str) -> Option<CounterType> {
+    let mut sum = CounterType::default();
+    for component in string.split(':') {
+        sum += component.parse::<CounterType>().ok()?;
```

Comment:
Not sure if a `sum.insert` or `sum.add` helper keeps the API cleaner and consistent with other distribution types.